ui/src/components/Listing/index.tsx

```diff
@@ -58,14 +59,58 @@ interface ListingProps {
   isFetching?: boolean;
   items: ListingItemProps[];
   Actions?: React.FC<{ id: string }>;
+  handlePagination?: any;
 }
 
+interface PaginationProps {
+  isNext: boolean;
+  isPrevious: boolean;
+  handleOffset: any;
+}
+
+const PaginationButtons: React.FC<PaginationProps> = (p) => (
+  <div className={c('pagination-buttons')}>
+    <button
+      className={c('load-button')}
+      onClick={() => p.handleOffset('pre')}
+      disabled={!p.isPrevious}
+    >
+      Previous
+    </button>
+
+    <button className={c('load-button')} onClick={() => p.handleOffset('next')} disabled={p.isNext}>
+      Next
+    </button>
+  </div>
+);
+
 const Listing: React.FC<ListingProps> = (p) => {
+  const [pagination, setPagination] = useState({ limit: 3, offset: 0 });
+  const items = p.items.slice(pagination.offset, pagination.offset + pagination.limit);
+  const isNext = p.items.length <= pagination.offset + pagination.limit;
+  const isPrevious = pagination.offset > 0;
+
+  const handleOffset = (mode: string) => {
+    const newOffset =
+      mode === 'pre' ? pagination.offset - pagination.limit : pagination.offset + pagination.limit;
+
+    setPagination({ ...pagination, offset: newOffset });
+
+    if (p.handlePagination) {
+      p.handlePagination(newOffset, pagination.limit);
```

Comment:
Let's pass an object in the argument here. It will save user of this function some searching for which argument means what.